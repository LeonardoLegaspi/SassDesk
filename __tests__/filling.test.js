var sass = require('node-sass');

describe('-----FILLING TEST', ()=>{
    test('Filling snapshot', async () => {
  
        var result = sass.renderSync({
          data: `
          @import './4_components/filling-button.scss';
          `,
          outputStyle: 'nested'
        });
        
        expect(result.css.toString()).toMatchSnapshot();
      });
  })