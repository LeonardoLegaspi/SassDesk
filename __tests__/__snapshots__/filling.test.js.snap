// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`-----FILLING TEST Filling snapshot 1`] = `
"/*
FIXME Is this useful? or limits to buttons with colors on Variables?
*/
.button, .button.tertiary-type, .filling-button button {
  border: 2px solid #bfbfbf;
  border-radius: 5px;
  display: inline-flex; }

.button, .button.tertiary-type, .filling-button button {
  cursor: pointer;
  background: transparent;
  color: #212120;
  font-size: 1rem;
  padding: 10px 30px;
  text-align: center;
  outline: none;
  transition: 300ms all; }
  .button.scheme, .filling-button button.scheme {
    border: 2px solid #1165ED;
    border-radius: 5px;
    background-color: #1165ED;
    color: #fff; }
    .button.scheme:hover:not(.inverted), .filling-button button.scheme:hover:not(.inverted) {
      border-color: #0e51bd;
      background: #0e51bd; }
      .button.scheme:hover:not(.inverted).empty, .filling-button button.scheme:hover:not(.inverted).empty {
        background: #fff;
        color: #1165ED; }
  .button.scheme-inverted, .filling-button button.scheme-inverted {
    border-color: #1165ED;
    background: #fff;
    color: #1165ED; }
    .button.scheme-inverted:hover, .filling-button button.scheme-inverted:hover {
      color: #fff;
      background: #1165ED;
      border-color: #1165ED; }
  .button.success, .filling-button button.success {
    border: 2px solid #46a93e;
    border-radius: 5px;
    background-color: #46a93e;
    color: #fff; }
    .button.success:hover:not(.inverted), .filling-button button.success:hover:not(.inverted) {
      border-color: #378430;
      background: #378430; }
      .button.success:hover:not(.inverted).empty, .filling-button button.success:hover:not(.inverted).empty {
        background: #fff;
        color: #46a93e; }
  .button.success-inverted, .filling-button button.success-inverted {
    border-color: #46a93e;
    background: #fff;
    color: #46a93e; }
    .button.success-inverted:hover, .filling-button button.success-inverted:hover {
      color: #fff;
      background: #46a93e;
      border-color: #46a93e; }
  .button.fail, .filling-button button.fail {
    border: 2px solid #d10000;
    border-radius: 5px;
    background-color: #d10000;
    color: #fff; }
    .button.fail:hover:not(.inverted), .filling-button button.fail:hover:not(.inverted) {
      border-color: #9e0000;
      background: #9e0000; }
      .button.fail:hover:not(.inverted).empty, .filling-button button.fail:hover:not(.inverted).empty {
        background: #fff;
        color: #d10000; }
  .button.fail-inverted, .filling-button button.fail-inverted {
    border-color: #d10000;
    background: #fff;
    color: #d10000; }
    .button.fail-inverted:hover, .filling-button button.fail-inverted:hover {
      color: #fff;
      background: #d10000;
      border-color: #d10000; }
  .button.warning, .filling-button button.warning {
    border: 2px solid #f1c40f;
    border-radius: 5px;
    background-color: #f1c40f;
    color: #212120; }
    .button.warning:hover:not(.inverted), .filling-button button.warning:hover:not(.inverted) {
      border-color: #c29d0b;
      background: #c29d0b; }
      .button.warning:hover:not(.inverted).empty, .filling-button button.warning:hover:not(.inverted).empty {
        background: #fff;
        color: #f1c40f; }
  .button.warning-inverted, .filling-button button.warning-inverted {
    border-color: #f1c40f;
    background: #fff;
    color: #f1c40f; }
    .button.warning-inverted:hover, .filling-button button.warning-inverted:hover {
      color: #212120;
      background: #f1c40f;
      border-color: #f1c40f; }
  .button.full-width, .filling-button button.full-width {
    width: 100%;
    display: inline-block; }
  .button.small, .filling-button button.small {
    font-size: 0.77rem; }
  .button.medium, .filling-button button.medium {
    font-size: 1.33rem; }
  .button.big, .filling-button button.big {
    font-size: 1.66rem; }
  .button.primary-type, .filling-button button.primary-type {
    border: 2px solid #1165ED;
    border-radius: 5px;
    background-color: #1165ED;
    color: #fff; }
    .button.primary-type:hover:not(.inverted), .filling-button button.primary-type:hover:not(.inverted) {
      border-color: #0e51bd;
      background: #0e51bd; }
      .button.primary-type:hover:not(.inverted).empty, .filling-button button.primary-type:hover:not(.inverted).empty {
        background: #fff;
        color: #1165ED; }
  .button.secondary-type, .filling-button button.secondary-type {
    border-color: #bfbfbf;
    background: #fff;
    color: #212120; }
    .button.secondary-type:hover, .filling-button button.secondary-type:hover {
      color: #fff;
      background: #212120;
      border-color: #212120; }
  .button:hover, .filling-button button:hover {
    border-color: #595959; }

.filling-button {
  width: min-content;
  position: relative;
  transition: 300ms all; }
  .filling-button button {
    background: transparent;
    border-radius: 0px;
    color: inherit; }
  .filling-button span {
    border: none;
    position: absolute;
    z-index: -1;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
    width: 0%;
    transition: 450ms all; }
  .filling-button:hover {
    color: #fff; }
    .filling-button:hover span {
      left: 0px;
      width: 100%; }
"
`;
